lib/middleware/csrfProtected.ts

```diff
@@ -4,27 +4,22 @@ import { getCsrfToken } from "next-auth/react";
 
 export const csrfProtected = (protectedMethods: string[]): MiddlewareRequest => {
   return async (req: NextApiRequest, res: NextApiResponse): Promise<MiddlewareReturn> => {
-    try {
-      if (isProtected(req, protectedMethods)) {
-        // Need to remove body from NextApiRequest due to Next-Auth bug
-        // Next-Auth sets the method Call based on if a Body is present on the request.
-        // This causes issues with CSRF protection so only required arguments are passed.
-        const clonedReq = { body: undefined, headers: req.headers };
+    if (isProtected(req, protectedMethods)) {
```

Comment:
After looking at the code I did not find a reason why this would have to be wrapped in a `try/catch` block

Reply:
I agree.  The only unknown is if the promise of `getCsrfToken` rejects.  After looking at next-auth source code we can see that they catch errors and return null.  If you really wanted to be safe in case they change that behaviour in the future you can add a .catch() to the promise.  This way if ever the next-auth code base behavior changes it won't cause an uncaught exception in the scope of your function.
```js
const csrfToken = await getCsrfToken({ req: clonedReq }).catch(() => null);
```